### PR TITLE
Visualization Update: AddMonitor method for Layer

### PR DESCRIPTION
**Pull request by harirakul**
Added test script `randMonitor.py` to monitor the `randInputs` test case

### CODE REVIEW

File: src/vivilux/visualize.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def __init__(self, name: str,
         self.ax.set_ylim(0, self.ylim)
         self.ax.legend(range(numLines))
 
+        plt.ion() # Enable interactive plot for real-time updates
```
Comment (NeuroSumbaD) [inline src/vivilux/visualize.py:46]: Is this call appropriate? If the user wants to add the monitor but do some pre-training without turning it on first, then the user should turn on and off interactive mode simultaneously. Maybe the layer class should have a function that calls a "switch on/off" function in the monitor class for each of its instances.

File: src/vivilux/layers.py [hunk 2]
```diff
@@ -174,6 +176,24 @@ def RunProcesses(self):
     def AddProcess(self, process: Process):
         process.AttachLayer(self)
 
+    def AddMonitor(self, monitor: Monitor):
+        self.monitors.append(monitor)
+
+    def UpdateMonitors(self):
+        self.UpdateSnapshot()
```
Comment (NeuroSumbaD) [inline src/vivilux/layers.py:183]: It appears `UpdateSnapshot` is called here and also in `StepTime`, is this intentional?

File: src/vivilux/layers.py [hunk 3]
```diff
@@ -174,6 +176,24 @@ def RunProcesses(self):
     def AddProcess(self, process: Process):
         process.AttachLayer(self)
 
+    def AddMonitor(self, monitor: Monitor):
+        self.monitors.append(monitor)
+
+    def UpdateMonitors(self):
```
Comment (NeuroSumbaD) [inline src/vivilux/layers.py:182]: If the monitor is added but turned off, it should not be updated. Should there be a boolean check here?

File: src/vivilux/layers.py [hunk 4]
```diff
@@ -174,6 +176,24 @@ def RunProcesses(self):
     def AddProcess(self, process: Process):
         process.AttachLayer(self)
 
+    def AddMonitor(self, monitor: Monitor):
+        self.monitors.append(monitor)
```
Comment (NeuroSumbaD) [inline src/vivilux/layers.py:180]: This was similar to my original implementation, but should we switch from a list to a dict (or both data structures is dict iterates faster)? That way we can turn on and off specific monitors by name.

**Review by NeuroSumbaD — APPROVED** [review]

File: src/vivilux/layers.py (hunk 2 above)
Comment (harirakul) [reply src/vivilux/layers.py:183]: No, that wasn't intentional. I forgot to delete `UpdateSnapshot` from `UpdateMonitors`.

File: src/vivilux/layers.py (hunk 3 above)
Comment (harirakul) [reply src/vivilux/layers.py:182]: I will add a `enable` boolean variable to the `Monitor` class in order to update the monitor only when `enable=True`.

File: src/vivilux/layers.py (hunk 4 above)
Comment (harirakul) [reply src/vivilux/layers.py:180]: That makes sense, I will store monitors in a dict instead.

**Pull request merged**